### PR TITLE
[No QA] Connect New Workspace Page with the backend

### PR DESCRIPTION
**Pull request by HorusGoul**
### Details
Removes stuff that won't ship in V1 and connects the form submission with the backend.

### Fixed Issues

Fixes https://github.com/Expensify/Expensify/issues/167378

### Tests
1. Open Expensify.cash
2. Go to the `/workspace/new` route
3. Introduce a name and press the "Get Started" button
4. Open the developer console, you should see this error:
image

### QA Steps
❌ 

### Tested On

- [x] Web
- [x] Mobile Web
- [x] Desktop
- [x] iOS
- [x] Android

#### Web
image

#### Mobile Web
image

#### Desktop
image
#### iOS
image
#### Android
image

### CODE REVIEW

File: src/libs/actions/Policy.js [hunk 1]
```diff
@@ -136,8 +140,35 @@ function invite(login, welcomeNote, policyID) {
         });
 }
 
+/**
+ * Merges the passed in login into the specified policy
+ *
+ * @param {String} name
+ */
+function create(name) {
+    Policy_Create({type: 'free', policyName: name})
```
Comment (marcaaron) [inline src/libs/actions/Policy.js:149]: We should use a `CONST` for this like `CONST.POLICY.TYPE`

File: src/libs/actions/Policy.js [hunk 2]
```diff
@@ -136,8 +140,35 @@ function invite(login, welcomeNote, policyID) {
...
+        .then((data) => {
+            if (data.jsonCode !== 200) {
+                // Show the user feedback
+                const errorMessage = translateLocal('workspace.new.genericFailureMessage');
+                Growl.show(errorMessage, CONST.GROWL.ERROR, 5000);
+
+                return;
+            }
+
+            Onyx.merge(ONYXKEYS.COLLECTION.POLICY + data.policyID, {
```
Comment (marcaaron) [inline src/libs/actions/Policy.js:159]:
```suggestion
            Onyx.merge(`${ONYXKEYS.COLLECTION.POLICY}${data.policyID}`, {
```

File: src/libs/actions/Policy.js [hunk 3]
```diff
@@ -136,8 +140,35 @@ function invite(login, welcomeNote, policyID) {
...
+ * @param {String} name
+ */
+function create(name) {
+    Policy_Create({type: 'free', policyName: name})
+        .then((data) => {
+            if (data.jsonCode !== 200) {
+                // Show the user feedback
+                const errorMessage = translateLocal('workspace.new.genericFailureMessage');
+                Growl.show(errorMessage, CONST.GROWL.ERROR, 5000);
+
```
Comment (marcaaron) [inline src/libs/actions/Policy.js:155]: unneeded `\n`

File: src/libs/API.js [hunk 4]
```diff
@@ -929,6 +929,17 @@ function GetCurrencyList() {
     return Mobile_GetConstants({data: ['currencyList']});
 }
 
+/**
+ * @param {object} parameters
+ * @param {string} [parameters.type]
+ * @param {string} [parameters.policyName]
```
Comment (marcaaron) [inline src/libs/API.js:935]: Capitalize the param types like we do elsewhere?

**Review by marcaaron — CHANGES_REQUESTED** [review]
LGTM and only have a few suggestions.

File: src/languages/es.js [hunk 5]
```diff
@@ -311,11 +311,9 @@ export default {
         new: {
             welcome: 'Bienvenido/a',
             chooseAName: 'Elige un nombre',
-            helpText: '¿Necesitas ayuda con la configuración? Pídenos una llamada y una persona de nuestro equipo te ayudará.',
+            helpText: '¡Dale un nombre a tu Workspace antes de activar tus Expensify Cards!',
             getStarted: '¡Empezar!',
-            uploadPhoto: 'Subir Foto',
-            editPhoto: 'Editar Foto',
-            requestCall: 'Concertar una llamada',
+            genericFailureMessage: 'Se ha producido un error al intentar crear el Workspace. Por favor intentalo de nuevo.',
```
Comment (Luke9389) [inline src/languages/es.js:316]: @HorusGoul do you speak Spanish or is this Google Translate? If it's the latter we should have a Spanish-speaking Expensifier verify that the grammar is good.

File: src/libs/actions/Policy.js (hunk 1 above)
Comment (HorusGoul) [reply src/libs/actions/Policy.js:149]: Done!

File: src/libs/actions/Policy.js (hunk 2 above)
Comment (HorusGoul) [reply src/libs/actions/Policy.js]: Done!

File: src/libs/actions/Policy.js (hunk 3 above)
Comment (HorusGoul) [reply src/libs/actions/Policy.js]: Done!

File: src/libs/API.js (hunk 4 above)
Comment (HorusGoul) [reply src/libs/API.js]: Done! I copied those from another comment, so I have also fixed these.

File: src/languages/es.js (hunk 5 above)
Comment (HorusGoul) [reply src/languages/es.js]: I'm a native speaker! Also, I just sent a commit fixing a typo.

**Review by marcaaron — APPROVED** [review]
LGTM!

**Comment by mvtglobally** [comment]
@marcaaron Should we halt QAing as this is `freePlan` beta ?

**Comment by marcaaron** [comment]
Yes please do not QA

**Pull request merged**